src/DynamoCoreWpf/Views/Debug/UpdateNodeIconsWindow.xaml.cs

```diff
@@ -450,6 +450,16 @@ private bool UpdateResxFile(NodeIconMetadata item, XDocument resx)
                     var oldElement = resxItems.Where(x => x.Attribute("name").Value.ToLower().Equals((item.IconName + "." + item.IconSuffix).ToLower())).FirstOrDefault();
                     if (oldElement != null)
                     {
+                        var currentElementValue = oldElement.Descendants("value").FirstOrDefault().Value;
+                        if (currentElementValue.ToLower().Contains(".png") || currentElementValue.ToLower().Contains(".jpg"))
+                        {
+                            //If the resx file is referencing a png image then we need to update/add the type and mimetype attributes
+                            XAttribute attRemove = oldElement.Attribute("type");
```

Comment:
@zeusongit I thought you mentioned there is no resx file referencing a legacy png, maybe I got it wrong?

Reply:
@QilongTang I've found 4 entries in DSCoreNodesImages.resx that are referencing the png image directly (without using base64).
https://github.com/DynamoDS/Dynamo/blob/2bec0090a87c3e7e18ae55507d31c3d28fcf6f7f/src/Libraries/CoreNodes/DSCoreNodesImages.resx#L7586

Reply:
@QilongTang I was wrong, there were 2 icons which were still using the png, we discussed this in [slack](https://autodesk.slack.com/archives/C01C2AGPSG2/p1717095656013579?thread_ts=1717016762.693159&cid=C01C2AGPSG2), roberto will remove these icons after updating them to b64.

Reply:
Just commited a change for removing the png files
https://github.com/DynamoDS/Dynamo/pull/15283/commits/6c591c46012e2aa03ff900c07b16c5988a360630

Reply:
Thanks, then the logic here is more for future proofing when things goes wrong from my understanding.